### DISCUSSION TITLE
Merge multiple pairs of images into single images

### DISCUSSION THREAD

**Discussion by AndromedusSoftware** [ImageMagick/ImageMagick#5275] [Help] [answered]
Hello.  I have two source folders called 'A' and 'B', and each of them contain lots of .png image files named 001.png, 002.png, 003.png etc.

What I want to do is merge A/001.png with B/001.png, A/002.png with B/002.png, and so on, and have the results saved into a third folder 'C'.  (When I say 'merge', I mean one image becomes a layer on top of the other and then they are flattened into a single layer.)
I would like the resulting merged images to be named the same as the source images, so they would become C/001.png, C/002.png, etc.

I have tried the following command in Windows command prompt:

magick A/\*.png B/\*.png -layers merge C/%d.png

This successfully merges two source images and places the result in the third folder, but unfortunately it only merges one of the pairs of images and names it 'C/0.png'.  I need it to do this for all of the images in the two source folders (which will actually be hundreds of images). I'm guessing it might actually be doing it for all of the source images, but it's probably overwriting the results with the same filename.  Clearly I'm doing something wrong!

Any suggestions please?  Thank you.

**Comment by snibgo** [comment]
I suggest you write a "magick" command that does what you want for a single pair of files. Then put that inside a "for" loop.

**Comment by AndromedusSoftware** [comment]
Ok, thank you.  I'd be grateful if you could elaborate though - what is the single command I should use that will give the merged file the same name as the source files?  And how do I do a for loop - do you mean like in a batch file ?

**Reply by fmw42** [reply]
Yes, batch file if on Windows. On Linux, etc, it would be a bash shell script..

Single command (assuming folders A,B,C exist and are at the same level in the current directory) 

Windows:

`magick A\001.png B\001.png -layers flatten C\001.png`

Unix:

`magick A/001.png B/001.png -layers flatten C/001.png`

**Reply by AndromedusSoftware** [reply]
Thank you.  I guess I'll need to go and learn how to write a batch file for this then.  Unless anyone here already knows what that would look like?

Cheers!

**Comment by snibgo** [comment]
It doesn't need to be a batch file.

**Comment by fmw42** [comment]
What is your IM version and platform/OS? Is the latter Windows?

As user snibgo said, it does not actually need to be a format batch file. You can just create the FOR loop over your images in the directories, use my command. Then just copy and paste the code to a CMD window to run it.

**Reply by AndromedusSoftware** [reply]
IM is the latest, I installed it today: 7.1.0-39
Windows version is Windows 10.

**Reply by fmw42** [reply]
Sorry, I cannot help further as I am on a Mac (Unix) and do not know Windows scripting.

**Reply by AndromedusSoftware** [reply]
No problem - thank you for your help so far.  I think I'm getting closer.

**Comment by AndromedusSoftware** [comment, marked as answer]
Just for completeness, I have now arrived at the solution which is the following batch file, which I saved as 'merge.bat' and run in the same directory as the three folders A, B and C.  Thanks for helping:

```
@echo off
FOR /R %%i IN (A\*.png) DO (
	
	echo merging %%~ni.png
	magick A\%%~ni.png B\%%~ni.png -layers flatten C\%%~ni.png
	
)
pause
```
